### PR TITLE
UUri Micro Serialization Implementation and Gherkin Tests

### PR DESCRIPTION
**Pull request by hostilechild007**
Created UUri Micro Serialization implementation in Python and Java Test Agents. 

- If user calls for Python/Java Test Agent to micro serialize a UUri Protobuf, then the respective Test Agent will return a byte serialized response 
- (similarly vice versa)
- Created Gherkin tests and Step Implementations to test above functionalities

Note: There are repeated commits in the history because they fall into the same commit intentions.

### CODE REVIEW

File: .github/workflows/tck-test.yml [hunk 1]
```diff
@@ -2,9 +2,9 @@ name: TCK Tests
 
 on:
   push:
-    branches: [ "main" ]
+    branches: [ "main", "microserialization"]
```
Comment (matthewd0123) [inline .github/workflows/tck-test.yml:5]: remove "microserialization" from here

File: .github/workflows/tck-test.yml [hunk 2]
```diff
@@ -2,9 +2,9 @@ name: TCK Tests
...
   pull_request:
-    branches: [ "main" ]
+    branches: [ "main", "microserialization"]
```
Comment (matthewd0123) [inline .github/workflows/tck-test.yml:7]: same as above

File: test_agent/java/src/main/java/org/eclipse/uprotocol/TestAgent.java [hunk 3]
```diff
@@ -28,19 +28,22 @@
 import com.google.protobuf.Any;
 import com.google.protobuf.Message;
 import com.google.protobuf.StringValue;
+// import org.eclipse.uprotocol.rpc.CallOptions;
```
Comment (matthewd0123) [inline test_agent/java/src/main/java/org/eclipse/uprotocol/TestAgent.java:31]: remove commented code

File: test_agent/python/constants.py [hunk 4]
```diff
@@ -23,6 +23,9 @@
 # SPDX-License-Identifier: Apache-2.0
 #
 # -------------------------------------------------------------------------
+from enum import Enum
```
Comment (matthewd0123) [inline test_agent/python/constants.py:26]: are these imports necessary?

File: test_agent/python/testagent.py [hunk 5]
```diff
@@ -48,13 +49,16 @@
...
 from uprotocol.uri.serializer.longuriserializer import LongUriSerializer
+from uprotocol.uri.serializer.microuriserializer import MicroUriSerializer
 from uprotocol.uuid.serializer.longuuidserializer import LongUuidSerializer
 from uprotocol.uuid.factory.uuidfactory import Factories
 from uprotocol.uri.validator.urivalidator import UriValidator
 
 import constants as CONSTANTS
 
-sys.path.append("../")
+# sys.path.append("../")
```
Comment (matthewd0123) [inline test_agent/python/testagent.py:59]: remove commented code

File: test_manager/features/steps/tck_step_implementations.py [hunk 6]
```diff
@@ -201,6 +201,7 @@ def verify_uuid_received_properties(context):
 def set_key_to_val(context: Context, key: str, value: str):
     if key not in context.json_dict:
         context.json_dict[key] = value
+    # context.logger.info(f"Dict -> {context.json_dict}")
```
Comment (matthewd0123) [inline test_manager/features/steps/tck_step_implementations.py:204]: remove commented code

File: test_manager/features/tests/serializers/micro_uri_serializer.feature
Comment (matthewd0123) [inline test_manager/features/tests/serializers/micro_uri_serializer.feature:1]: align this file with the formatting of long_uri_deserializer. We are no longer specifying uE1, instead specifying it when we run the behave script

File: test_manager/features/tests/serializers/micro_uri_deserializer.feature
Comment (matthewd0123) [inline test_manager/features/tests/serializers/micro_uri_deserializer.feature:1]: align this file with the formatting of long_uri_deserializer. We are no longer specifying uE1, instead specifying it when we run the behave script

**Review by neelam-kushwah — APPROVED** [review]

**Pull request merged**